lib/connection/command_result.js

```diff
@@ -19,7 +19,9 @@ var CommandResult = function(result, connection, message) {
  * @return {object}
  */
 CommandResult.prototype.toJSON = function() {
-  return this.result;
+  let result = this;
```

Comment:
I think we should be making a copy here. Deleting the message b/c of a toJSON call seems like an unintended side effect.

Reply:
Good catch, thanks!